apps/files/ajax/newfile.php

```
@@ -61,6 +61,15 @@
 	exit();
 }
 
+if (\OC\Files\Filesystem::isForbiddenFileOrDir($fileName)) {
```

What's about "upload then? There you also get a generic error message only?
Maybe it would be a good idea to let the hook throw an exception with the corresponding error message if it is forbidden and then make sure that we display the error message from the exception instead of a generic one when we catch the exception. You could use the HintException to provide a translated message. We do something similar with encryption errors.

It is probably more failure save to add if statements all over the place and it is more generic since other errors can be handled the same way.